packages/camera/camera_platform_interface/test/types/camera_image_test.dart

```diff
@@ -0,0 +1,113 @@
+      CameraImage cameraImage = CameraImage.fromPlatformData(<dynamic, dynamic>{
+        'format': 35,
+        'height': 1,
+        'width': 4,
+        'planes': [
+          {
+            'bytes': Uint8List.fromList([1, 2, 3, 4]),
+            'bytesPerPixel': 1,
+            'bytesPerRow': 4,
+            'height': 1,
+            'width': 4
+          }
+        ]
+      });
+      expect(cameraImage.height, 1);
+      expect(cameraImage.width, 4);
+      expect(cameraImage.format.group, ImageFormatGroup.yuv420);
+      expect(cameraImage.planes.length, 1);
+    });
+
+    test('$CameraImage has ImageFormatGroup.yuv420 for iOS', () {
+      debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+      CameraImage cameraImage = CameraImage.fromPlatformData(<dynamic, dynamic>{
+        'format': 875704438,
+        'height': 1,
+        'width': 4,
+        'planes': [
+          {
+            'bytes': Uint8List.fromList([1, 2, 3, 4]),
+            'bytesPerPixel': 1,
+            'bytesPerRow': 4,
+            'height': 1,
+            'width': 4
+          }
+        ]
+      });
+      expect(cameraImage.format.group, ImageFormatGroup.yuv420);
+    });
+
+    test('$CameraImage has ImageFormatGroup.yuv420 for Android', () {
+      debugDefaultTargetPlatformOverride = TargetPlatform.android;
+
+      CameraImage cameraImage = CameraImage.fromPlatformData(<dynamic, dynamic>{
+        'format': 35,
+        'height': 1,
+        'width': 4,
+        'planes': [
+          {
+            'bytes': Uint8List.fromList([1, 2, 3, 4]),
+            'bytesPerPixel': 1,
+            'bytesPerRow': 4,
+            'height': 1,
+            'width': 4
+          }
+        ]
+      });
+      expect(cameraImage.format.group, ImageFormatGroup.yuv420);
+    });
+
+    test('$CameraImage has ImageFormatGroup.bgra8888 for iOS', () {
+      debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+
+      CameraImage cameraImage = CameraImage.fromPlatformData(<dynamic, dynamic>{
+        'format': 1111970369,
+        'height': 1,
+        'width': 4,
+        'planes': [
+          {
+            'bytes': Uint8List.fromList([1, 2, 3, 4]),
+            'bytesPerPixel': 1,
+            'bytesPerRow': 4,
+            'height': 1,
+            'width': 4
+          }
+        ]
+      });
+      expect(cameraImage.format.group, ImageFormatGroup.bgra8888);
+    });
+    test('$CameraImage has ImageFormatGroup.unknown', () {
+      CameraImage cameraImage = CameraImage.fromPlatformData(<dynamic, dynamic>{
+        'format': null,
+        'height': 1,
+        'width': 4,
+        'planes': [
+          {
+            'bytes': Uint8List.fromList([1, 2, 3, 4]),
+            'bytesPerPixel': 1,
+            'bytesPerRow': 4,
+            'height': 1,
+            'width': 4
+          }
+        ]
+      });
+      expect(cameraImage.format.group, ImageFormatGroup.unknown);
+    });
+  });
+}
```

Comment:
Test for the `CameraImage` class can also be moved to the `camera/camera` package.